src/test/java/com/wikia/webdriver/common/core/elemnt/Wait.java

```diff
@@ -395,7 +396,30 @@ public String toString() {
     }
   }
 
-  public void forUrlContains(String text){
+  public void forSuccessfulResponseByUrlPattern(final NetworkTrafficInterceptor trafficInterceptor,
+                                                final String pattern) {
+    changeImplicitWait(0, TimeUnit.SECONDS);
+    try {
+      wait.until(
+          new ExpectedCondition<Boolean>() {
+            @Override
+            public Boolean apply(WebDriver webDriver) {
+              HarEntry entry = trafficInterceptor.getMatchingEntryByUrlPatternFromHar(pattern);
+              return entry != null && entry.getResponse().getStatus() < 400;
+            }
+
+            @Override
+            public String toString() {
+              return "successful response matching pattern: " + pattern;
```

Comment:
![screen shot 2016-05-13 at 09 43 29](https://cloud.githubusercontent.com/assets/2042649/15241313/6a25680e-18ef-11e6-9dfe-8604b6050c4a.png)

Does it look fine? :)